### ISSUE TITLE
Use offset from SDK for executable loaders instead of re-calculating it

### ISSUE THREAD

**Issue by kayagokalp** [FuelLabs/sway#6683]
## Description

Loader offset is exposed by sdk as of https://github.com/FuelLabs/fuels-rs/pull/1522 currently we are recalculating this value in forc-deploy because at the time of the implementation this was not exposed by sdk. This is bad because if anything changes around that area, we need to update it manually. We should use the exposed value from sdk instead of recalculating.

https://github.com/FuelLabs/fuels-rs/pull/1522

### Context
-------
https://github.com/FuelLabs/sway/blob/b6bbbf8bf5939eaf2c965b4b94732d43ef656500/forc-plugins/forc-client/src/op/deploy.rs#L357-L360

## Solution

Basically remove the function showed in context, and get the exposed value [from SDK](https://github.com/FuelLabs/fuels-rs/pull/1522) directly.

**Comment by Abeeujah** [comment]
Hi, this sounds great, I'd love to jump on this!

**Comment by kayagokalp** [comment]
> Hi, this sounds great, I'd love to jump on this!

Hey @Abeeujah go for it! Let me know if you need any help/guidance.

**Comment by Abeeujah** [comment]
Hi @kayagokalp the function signature of the sdk is different from the one in deploy.rs
forc-plugins/forc-client/src/op/deploy.rs
```rust
fn loader_data_offset(binary: &[u8], blob_id: &BlobId) -> Result<Option<usize>> {
```
packages/fuels-core/src/lib.rs
```rust
pub fn with_shifted_offsets(self, shift: i64) -> Result<Self> {
```

Am I looking at the right functions? or there's another I'm supposed to look at?

**Comment by Abeeujah** [comment]
Hi @kayagokalp the function signature of the sdk is different from the one in deploy.rs
forc-plugins/forc-client/src/op/deploy.rs
```rust
fn loader_data_offset(binary: &[u8], blob_id: &BlobId) -> Result<Option<usize>> {
```
packages/fuels-core/src/lib.rs
```rust
pub fn with_shifted_offsets(self, shift: i64) -> Result<Self> {
```

Am I looking at the right functions? or there's another I'm supposed to look at?

**Comment by Abeeujah** [comment]
Hi @kayagokalp the function signature of the sdk is different from the one in deploy.rs
forc-plugins/forc-client/src/op/deploy.rs
```rust
fn loader_data_offset(binary: &[u8], blob_id: &BlobId) -> Result<Option<usize>> {
```
packages/fuels-core/src/lib.rs
```rust
pub fn with_shifted_offsets(self, shift: i64) -> Result<Self> {
```

Am I looking at the right functions? or there's another I'm supposed to look at?
